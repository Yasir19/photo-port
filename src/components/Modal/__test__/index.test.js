import React from 'react'
import { render, cleanup, fireEvent } from'@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Modal from '..'

afterEach(cleanup)
const currentPhoto ={ 
    name: 'Park bench',
    category: 'landscape',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    index:1
};

const mockToggleModal = jest.fn();


describe('content componenet', () => {
    it('render', () => {
        render(<Modal
        onClose={mockToggleModal}
        currentPhoto={currentPhoto}
        />)
    })
    it('matches snapshot Dom node structure', () => {
        const { asFragment } = render(<Modal
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
        />)
        expect(asFragment).toMatchSnapshot()
    })
})

describe('Click Event', () => {
    it('calls onClose handler', () => {
 const {getByText} = render(<Modal
 onClose={mockToggleModal}
 currentPhoto={currentPhoto}
 />)
 fireEvent.click(getByText('Close the Picture'));
 expect(mockToggleModal).toHaveBeenCalledTimes(1)
    });
  }) 
