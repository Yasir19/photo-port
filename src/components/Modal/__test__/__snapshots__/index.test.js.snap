// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content componenet matches snapshot Dom node structure 1`] = `[Function]`;
